src/software/embedded/yaml/yaml_reader.cpp

```diff
@@ -0,0 +1,16 @@
+#include "software/logger/logger.h"
+#include "software/embedded/platform.h"
+#include "software/embedded/constants/constants.h"
+#include "yaml_reader.hpp"
+
+
+YamlReader::YamlReader(const char* path_to_file)
+    : node_(loadNode(path_to_file)) { }
+
+YAML::Node YamlReader::loadNode(const char* path_to_file){
+    if constexpr (PLATFORM == Platform::LIMITED_BUILD)
+        return YAML::Load(default_yaml); 
```

Comment:
```suggestion
    if constexpr (PLATFORM == Platform::LIMITED_BUILD)
    {
        return YAML::Load(default_yaml); 
    }
```
explicit braces are part of our coding style guide: https://github.com/UBC-Thunderbots/Software/blob/3c8ee82dcdc60339afa84646d5b1c62330d92bed/docs/code-style-guide.md?plain=1#L307